File: Labwork 4/Python/Part1.py
# 1.
a = 5          # int
b = "5"        # str
c = 5.0        # float

# 2.
print(a + c)          
print(a + b)          
print(b + str(c))     


print(a + int(b))     
print(float(a) + c)   
print(str(a) + b)   

# 3.
# 1) Which fail and why?
#   a + b fails because Python does not implicitly convert between numbers and strings.
# 2) Which language allows implicit coercion and how safely?
#   Python only coerces compatible numeric types (int ↔ float), not strings.
# 3) What does this tell you about strong vs weak typing?
#   Python is strongly typed and dynamically typed — conversions must be explicit.

File: Labwork 4/C#/Part1.cs

using System;

class Program
{
    static void Main()
    {
        int a = 5;
        string b = "5";
        double c = 5.0;

        Console.WriteLine(a + c);   
        Console.WriteLine(a + b);   
        Console.WriteLine(b + c);   

        Console.WriteLine(a + Convert.ToInt32(b));  
        Console.WriteLine((double)a + c);           
        Console.WriteLine(a.ToString() + b);        
    }
}



// • Which of these operations fail and why?
//    None fail. 'a + b' works due to operator overloading for string concatenation.
// • Which language allows implicit coercion and how safely?
//    C# safely coerces numeric types (int ↔ double) and allows string concatenation.
//      Implicit conversions are safe and checked by the compiler.
// • What does this tell you about strong vs weak typing?
//    C# is strongly and statically typed.
//      It allows limited, controlled implicit conversions.
